prow/plugins/updateconfig/updateconfig_test.go

```diff
@@ -288,133 +492,31 @@ func TestUpdateConfig(t *testing.T) {
 			if !strings.Contains(comment, "Updated the") {
 				t.Errorf("%s: missing Updated the from %s", tc.name, comment)
 			}
-			for _, configName := range tc.configUpdates {
-				if !strings.Contains(comment, configName) {
-					t.Errorf("%s: missing %s from %s", tc.name, configName, comment)
+			for configName := range tc.updatedConfigMaps {
+				found := false
+				for _, name := range fkc.updatedMaps {
```

Comment:
This validates that every expected update is actually made. We already check this below when we loop over `updatedConfigMaps`. Here we need to validate that every actual update is expected.

Reply:
hummmm, nope? `updatedConfigMaps` can contain configmaps that's not updated

Reply:
Consider the case in which there is a bug in the config updater that makes it always add a configmap named `unknown`. The tests should identify that we updated a configmap that wasn't listed in `tc.updatedConfigMaps`. Currently, we are checking that the known configmaps are updated properly, but we aren't checking that unexpected configmaps are not updated.

> updatedConfigMaps can contain configmaps that's not updated

It should probably be renamed then :smile:

Reply:
renamed, and added another reverse compare loop

Reply:
LGTM. The new test caught a bug in the test logic though. (The merge SHA should only be set if it is specified in the test case).

Reply:
forgot to push, fixed